crates/distribution-types/src/file.rs

```diff
@@ -51,7 +53,12 @@ impl File {
             size: file.size,
             upload_time_utc_ms: file.upload_time.map(|dt| dt.timestamp_millis()),
             url: if file.url.contains("://") {
-                FileLocation::AbsoluteUrl(file.url)
+                let url = safe_copy_url_auth(
```

Comment:
Is it possible to skip this work if there's no username or password on the `base`?

Reply:
Yeah we should be able to avoid it in that case

Reply:
Skipped in https://github.com/astral-sh/uv/pull/1886/commits/eb1d6a611fab61dfeedf462ae1baa3d763cdabdd

No changes to integration benchmarks